presto-main/src/test/java/com/facebook/presto/operator/aggregation/AggregationTestUtils.java

```diff
@@ -238,22 +262,24 @@ public static Object partialAggregation(InternalAggregationFunction function, do
     public static Object partialAggregation(InternalAggregationFunction function, double confidence, int[] args, Page... pages)
     {
         AccumulatorFactory factory = function.bind(Ints.asList(args), Optional.empty(), Optional.empty(), confidence);
-        Accumulator partialAggregation = factory.createAccumulator();
+        Accumulator finalAggregation = factory.createIntermediateAccumulator();
+
+        // Test handling of empty intermediate blocks
+        Accumulator emptyAggregation = factory.createAccumulator();
+        Block emptyBlock = getIntermediateBlock(emptyAggregation);
+
+        finalAggregation.addIntermediate(emptyBlock);
+
         for (Page page : pages) {
+            Accumulator partialAggregation = factory.createAccumulator();
             if (page.getPositionCount() > 0) {
```

Comment:
This is different. You can have empty intermediate state. But you can't call `addInput` with an empty page on a state.